src/transformers/modeling_utils.py

```diff
@@ -2071,10 +2088,15 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                     " bitsandbytes `pip install -i https://test.pypi.org/simple/ bitsandbytes` or"
                     " pip install bitsandbytes` "
                 )
-            if torch_dtype == "auto" or torch_dtype != torch.float16:
+            if torch_dtype != torch.float16:
                 # We force the `dtype` to be float16, this is a requirement from `bitsandbytes`
+                logger.warning(
+                    f"Overriding torch_dtype={torch_dtype} with `torch_dtype=torch.float16` due to "
+                    "requirements of `bitsandbytes` to enable model loading in mixed int8. "
+                    "Either pass torch_dtype=torch.float16 or don't pass this argument at all to remove this warning."
+                )
                 torch_dtype = torch.float16
-                logger.info("Loading the model in mixed int8 - forcing the weights to be casted in float16")
+
```

Comment:
can't swipe such a hijacking of users args into `logger.info` - this is a warning material. if anybody were to read it.

I think it'd work better to assert here, since if the user went through the pain of asking for a specific dtype we should tell aloud that their request can't be satisfied and as you know warnings ain't it. But this is BC material so using warning.

Reply:
cc @younesbelkada